File: src/MainService.Models/Entities/Address.cs
namespace MainService.Models.Entities
{
    public class Address
    {
        public int Id { get; set; }
        public string? Name { get; set; }
        public string? Description { get; set; }
        public string? Street { get; set; }
        public string? InteriorNumber { get; set; }
        public string? ExteriorNumber { get; set; }
        public string? Neighborhood { get; set; }
        public string? City { get; set; }
        public string? State { get; set; }
        public string? Country { get; set; }
        public string? Zipcode { get; set; }
        public string? Notes { get; set; }
        public string? CrossStreet1 { get; set; }
        public string? CrossStreet2 { get; set; }
        public double? Latitude { get; set; }
        public double? Longitude { get; set; }
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

        public ClinicLogo ClinicLogo { get; set; } = null!;
        public DoctorClinic DoctorClinic { get; set; } = null!;
        public List<EventClinic> EventClinics { get; set; } = [];
        public List<ClinicNurse> ClinicNurses { get; set; } = [];
        public List<OrderDeliveryAddress> OrderDeliveryAddresses { get; set; } = [];
        public List<OrderPickupAddress> OrderPickupAddresses { get; set; } = [];
        public UserAddress UserAddress { get; set; } = null!;
        public List<PrescriptionClinic> PrescriptionClinics { get; set; } = [];

        public double GetLatitude() => Latitude ?? 0;
        public double GetLongitude() => Longitude ?? 0;

        public string? GetPhotoUrl() {
            if (ClinicLogo != null &&
                ClinicLogo.Photo != null &&
                ClinicLogo.Photo.Url != null &&
                !string.IsNullOrEmpty(ClinicLogo.Photo.Url.AbsoluteUri))
            {
                return ClinicLogo.Photo.Url.AbsoluteUri;
            }

            return null;
        }
    }
}

File: client/src/app/_models/addresses/address.ts
import { Addresses } from "src/app/_models/addresses/addressTypes";
import { Entity } from "src/app/_models/base/entity";
import { SelectOption } from "src/app/_models/base/selectOption";


export class Address extends Entity {
  street: string | null = null;
  exteriorNumber: string | null = null;
  interiorNumber: string | null = null;
  neighborhood: string | null = null;
  city: string | null = null;
  state: string | null = null;
  country: string | null = null;
  zipcode: string | null = null;
  photoUrl: string | null = null;
  latitude: number | null = null;
  longitude: number | null = null;
  type: Addresses | null = null;

  nursesCount: number | null = null;
  isMain: boolean | null = null;

  select: SelectOption | null = null;

  constructor(init?: Partial<Omit<Address, 'address'>>) {
    super();

    Object.assign(this, init);
  }

  get address(): string {
    return `${this.street} ${this.exteriorNumber} ${this.interiorNumber}, ${this.neighborhood}, ${this.city}, ${this.state}, ${this.country}, ${this.zipcode}`;
  }
}
